From 03ce0295e021c831816707c176c6eb72705dcf9b Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Fri, 22 Sep 2017 12:11:22 +0800
Subject: [PATCH 2563/5242] MLK-16526-2 thermal: qoriq: add buffer for passive
 cooling mechanism

commit  4f85b4294a0d86d1a6bcc44489c9d89ad23e1ab2 from
https://source.codeaurora.org/external/imx/linux-imx.git

On i.MX8MQ, When temperature exceeds passive point,
the cooling mechanism will be trigger and temperature
will begin to drop, to avoid back and forth surrounding
the passive point, here adds 10 C buffer for passive point,
that means when cooling mechanism is trigger, only after
the temperature drop to 10 C below the passive point,
the cooling mechanism will exit.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/thermal/qoriq_thermal.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/thermal/qoriq_thermal.c b/drivers/thermal/qoriq_thermal.c
index c6767de..c832ee8 100644
--- a/drivers/thermal/qoriq_thermal.c
+++ b/drivers/thermal/qoriq_thermal.c
@@ -26,6 +26,8 @@
 
 #define SITES_MAX	16
 
+#define TMU_TEMP_PASSIVE_COOL_DELTA	10000
+
 /*
  * QorIQ TMU Registers
  */
@@ -211,7 +213,8 @@ static int tmu_get_trend(void *p,
 	trip_temp = (trip == TMU_TRIP_PASSIVE) ? data->temp_passive :
 					     data->temp_critical;
 
-	if (data->tz->temperature >= trip_temp)
+	if (data->tz->temperature >=
+		(trip_temp - TMU_TEMP_PASSIVE_COOL_DELTA))
 		*trend = THERMAL_TREND_RAISE_FULL;
 	else
 		*trend = THERMAL_TREND_DROP_FULL;
-- 
1.7.9.5

